BootloaderCorePkg/Library/MpInitLib/Ia32/MpInitLib.c

```diff
@@ -30,17 +35,26 @@ UINT32                             mMpInitPhase = EnumMpInitNull;
 VOID
 SmmRebase (
   IN UINT32  Index,
-  IN UINT32  ApicId
+  IN UINT32  ApicId,
+  IN  INT32  Offset
 )
 {
-  INT32   Offset;
+  UINT32  SmmBase;
+
+  if (Offset == 0) {
+    Offset = PcdGet32 (PcdSmramTsegSize) - (SMM_BASE_MIN_SIZE + Index * SMM_BASE_GAP);
+    if (Offset <= 0) {
+      return;
+    }
+  }
 
-  Offset = PcdGet32 (PcdSmramTsegSize) - (SMM_BASE_MIN_SIZE + Index * SMM_BASE_GAP);
-  if (Offset <= 0) {
-    return;
+  SmmBase = PcdGet32 (PcdSmramTsegBase) + Offset;
```

Comment:
If we have to convert from offset to base, then can we just pass in the base directly in the SmmRebase() function ?

Reply:
Agree. Makes it easier . Did the change in new patch.